databricks/koalas/indexes.py

```diff
@@ -1278,6 +1282,81 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def delete(self, loc):
+        """
+        Make new Index with passed location(-s) deleted.
+
+        .. note:: this API can be pretty expensive since it is based on
+             a global sequence internally.
+
+        Returns
+        -------
+        new_index : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
+        >>> kidx
+        Int64Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete(0)
+        Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete([0, 1, 2, 3, 10, 11])
+        Int64Index([4, 2, 4, 4, 2, 2], dtype='int64')
+        """
+        if isinstance(self, MultiIndex):
+            raise NotImplementedError("currently doesn't support for MultiIndex")
+        if not is_list_like(loc):
+            if not isinstance(self, str) and abs(loc) >= len(self):
+                raise IndexError(
+                    "index {} is out of bounds for axis 0 with size {}".format(loc, len(self))
+                )
+            loc = [loc]
+        loc = [int(item) for item in loc]
+        loc = [item if item >= 0 else len(self) + item for item in loc]
+
+        # we need below temporal column 'index_value_columns'
+        # since '_InternalFrame.attach_default_index' will be failed
+        # when if self._scol has name of '__index_level_0__'
+        index_value_column = "__index_value__"
```

Comment:
I think you should use `verify_temp_column_name` to verify the temp column name

Reply:
Oh I didn't realize that methods. thanks :)